modules/command-engine/core/src/main/scala/surge/internal/kafka/KafkaProducerActorImpl.scala

```diff
@@ -203,9 +231,11 @@ class KafkaProducerActorImpl(
     case msg: ProducerFenced => maybeRestartFencedProducer(msg)
     case msg: EventsFailedToPublish =>
       context.become(fenced(state.completeTransaction(), initialFenceTime))
-      msg.originalSenders.foreach(_ ! KafkaProducerActor.PublishFailure(msg.reason))
-    case RestartProducer              => restartPublisher()
-    case ShutdownProducer             => stopPublisher()
+      msg.originalSenders.foreach(sender => sender.actorRef ! KafkaProducerActor.PublishFailure(sender.trackingId, msg.reason))
+    case RestartProducer  => restartPublisher()
+    case ShutdownProducer => stopPublisher()
+    case ClearExpiredTrackers =>
```

Comment:
We should handle this in the other states for this actor as well to avoid dead letters from the scheduled message not being handled since we're kinda expecting them to show up all the time in any state. Handling can just be ignoring it:
```
case ClearExpiredTrackers => // Ignore and do nothing
```

Reply:
done